zip-0320.rst

```diff
@@ -81,50 +80,78 @@ Requirements
-=============
+   and decoding should be required as a consequence of this ZIP. In particular,
+   conversion between a transparent-source-only address and the corresponding
+   unrestricted transparent address should be possible using only dependencies
+   that are available to Binance's front-end code.
 
-This alternative was suggested by @hanh in [#hanh-suggestion]_.
+It is only required to support a transparent-source-only form of P2PKH addresses;
+P2SH address support is not necessary.
 
-TEX Addresses
--------------
-
-A TEX Address is a Bech32m [#bip-0350]_ reencoding of a transparent Zcash
-P2PKH address [#protocol-transparentaddrencoding]_.
 
-Motivations for Alternative 1
------------------------------
+Specification
+=============
 
-The TEX Address is the simplest possible approach to creating a new address
-type that indicates that only transparent sources of funds should be used.
+A TEX Address, also called a Transparent-Source-Only Address, is a Bech32m [#bip-0350]_
+reencoding of a transparent Zcash P2PKH address [#protocol-transparentaddrencoding]_.
 
-Specification (Alternative 1)
------------------------------
+Wallets and other Senders sending to a TEX address (as any output) MUST ensure that
+only transparent (P2SH or P2PKH) UTXOs are spent in the creation of the transaction.
+For simplicity of parsing and interpreting such transactions, they also SHOULD only
+send to transparent outputs.
 
-A TEX address is produced from a Mainnet Zcash P2PKH Address by executing the
+A TEX address can be produced from a Mainnet Zcash P2PKH Address by executing the
 following steps:
 
 1. Decode the address to a byte sequence using the Base58Check decoding
    algorithm [#Base58Check]_.
 2. If the length of the resulting byte sequence is not 22 bytes or if its two-byte
-   address prefix is not :math:`[\mathtt{0x1C}, \mathtt{0xB8}]`, return an error. Otherwise,
-   let the **validating key hash** be the remaining 20 bytes of the sequence after
-   removing the two-byte address prefix.
+   address prefix is not :math:`[\mathtt{0x1C}, \mathtt{0xB8}]`, return an error.
+   Otherwise, let the **validating key hash** be the remaining 20 bytes of the
+   sequence after removing the two-byte address prefix.
 3. Reencode the 20-byte **validating key hash** using the Bech32m encoding
    defined in [#bip-0350]_ with the human-readable prefix (HRP) ``"tex"``.
 
 For Testnet addresses, the required lead bytes of a P2PKH address in step 2 are
 :math:`[\mathtt{0x1D}, \mathtt{0x25}]`, and the ``"textest"`` HRP is used when
 reencoding in step 3.
 
-Wallets and other Senders sending to a TEX address (as any output) MUST
-ensure that only transparent UTXOs are spent in the creation of a
-transaction.
+A TEX address can be parsed by reversing this encoding, i.e.:
+
+1. Decode the address to a byte sequence using Bech32m [#bip-0350]_, checking
+   that the HRP is ``"tex"`` for a Mainnet TEX Address and ``"textest"`` for a
+   Testnet TEX Address.
```

Comment:
Since this is Bech32m, we also have support for "texregtest" in `zcash_address`, should that be mentioned here?

Reply:
In general regtest is considered an implementation detail of zcashd that doesn't need to be considered in ZIPs or the protocol spec.